src/pipx/commands/common.py

```diff
@@ -116,6 +153,37 @@ def _symlink_package_apps(
             )
 
 
+def find_selected_venvs_for_package(
+    venv_container: VenvContainer, package: str, *, verbose=False
+) -> List[Venv]:
+    """
+    Returns all venvs that are selected as default for a package. If all venv metadata is valid, then only one venv is
+    selected. However, this function still searches for multiple ones to be able to recover from an invalid state.
+    """
+    selected_venvs = []
+
+    for venv_dir in venv_container.iter_venv_dirs():
+        # optimization: don't check packages that are not of the format f'{package}{suffix}'
+        if not venv_dir.name.startswith(package):
+            continue
```

Comment:
Could be removed, though generally optimizations are effective if they reduce the number of filesystem interactions.